### DISCUSSION TITLE
A friendlier Notes editor?

### DISCUSSION THREAD

**Discussion by leekelleher** [leekelleher/umbraco-contentment#187] [Idea]
The configuration of the [Notes editor](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/notes.md) is quite basic. Restricted to the usage of the rich-text editor.

With my own usage of notes for content-editors, I like to add a bit more presentation. e.g. within a panel, often with an icon. To give the note more of visual impact.

For some of my own client projects, I have extended the Contentment Notes editor with the following configuration options...

![contentment--editor-note--config](https://user-images.githubusercontent.com/209066/144250863-0b591953-0e6e-4d8a-83ab-affb204c085a.png)

The resulting editor note would be displayed as...

![contentment--editor-note--render](https://user-images.githubusercontent.com/209066/144250878-01bf6eb4-b284-44bd-9dd2-6dd60a4e64cf.png)

In terms of Contentment, I am wondering whether this could either be a replacement for the Notes editor? or an additional editor, e.g. Notes 2 or "Editor Note"?

**Comment by skttl** [comment]
I like it! And would probably use it - especially for things where you need to refactor doctypes, while still using them.

Like if you had a doc type with grid+dtge, but you wanted to switch to something else. You need the grid+dtge content until all nodes have been converted by the editors, so this note can be a good way of telling the editor to stop editing the grid content, and start migrating to newer options in the doc type.

Combined with the EditorModelEventManager it could be shown/hidden based on different circumstances.

**Reply by leekelleher** [reply]
Thanks @skttl. Good use-case too!

**Comment by CarlSargunar** [comment]
Definitely a feature we'll use on live sites, and sites that are up for a while. 

I realise it's probably beyond the scope of what you're trying to do with Contentment, but to go along with the ability to show or hide a label, if we had the ability to show or hide an entire property group while it's being worked on that could be useful

New features aside - definitely useful and one the product manager on my client's site will love

**Reply by leekelleher** [reply]
Thanks @CarlSargunar, good suggestion.
I can't think off-the-top of my head how that'd work, still something to chew on in the back of my brain.  🤔😆

**Comment by NikRimington** [comment]
I like it, have you looked at EditorNotes package from Marc? It's very similar to this bit has a couple of other options.

Make the notes collapsible with different start states, or make them a toggle that opens a dedicated dialogue.

 Just food for thought.

**Reply by leekelleher** [reply]
Thanks @NikRimington. I have much love for EditorNotes. (I mention it in [the docs for Contentment's Notes](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/notes.md). 😁)

I hadn't considered its features for this, (not that I'm aiming to complete with EditorNotes), but yeah could make sense. 👍

**Comment by leekelleher** [comment]
Implemented in PR #204 - aiming to include in upcoming Contentment v3.2.0.
